functions.global.php

```diff
@@ -21,6 +21,108 @@
 	exit;
 }
 
+/**
+ * Hook into Core's _deprecated_function
+ * Add more details about when a deprecated function will be removed.
+ *
+ * @since 8.8.0
+ *
+ * @param string $function    The function that was called.
+ * @param string $replacement Optional. The function that should have been called. Default null.
+ * @param string $version     The version of Jetpack that deprecated the function.
+ */
+function jetpack_deprecated_function( $function, $replacement, $version ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+	// Bail early for non-Jetpack deprecations.
+	if ( 0 !== strpos( $version, 'jetpack-' ) ) {
+		return;
+	}
+
+	// Look for when a function will be removed based on when it was deprecated.
+	$removed_version = jetpack_get_future_removed_version( $version );
+
+	// If we could find a version, let's log a message about when removal will happen.
+	if (
+		! empty( $removed_version )
+		&& ( defined( 'WP_DEBUG' ) && WP_DEBUG )
```

Comment:
Should we respect the `deprecated_function_trigger_error` Core filter? Would it be weird to have this in the error log without the deprecation notice or despite the choice not to issue those warnings?

Reply:
Good point. Done in 51fa1b1af2f8a6b3be4b9a44b4ea3ed7530fa58e